### PR TITLE
Add the global nesting capability

### LINKED ISSUE #2544: Add the global nesting capability
### What new functionality do you need?

Run a regional domain nested inside a global run

### What are the requirements for the new functionality?

The Atmospheric River project requires to run the global workflow with a nested domain

### Acceptance Criteria

Be able to run the global workflow with a nested domain

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by guoqing-noaa**
# Description

This PR adds the global nesting capability to the global workflow so that one can run a regional domain nested in a global run.

Resolves #2544 

# Type of change

- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
- Clone and build on Hera
- Forecast-only on Hera

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: ush/parsing_model_configure_FV3.sh [hunk 1]
```diff
@@ -59,6 +59,27 @@ if [[ ! -f ${template} ]]; then
...
+cen_lat:                 ${target_lat}
+lon1:                    ${NEST_LON1}
+lat1:                    ${NEST_LAT1}
+lon2:                    ${NEST_LON2}
+lat2:                    ${NEST_LAT2}
+dlon:                    ${NEST_DLON}
+dlat:                    ${NEST_DLAT}
+</output_grid_02>
+EOF
+fi
```
Comment (aerorahul) [inline ush/parsing_model_configure_FV3.sh]: We should not do this.  The ufs-weather-model should have a `model_configure` for `nest` case, that should be used.

File: parm/config/gfs/config.resources [hunk 2]
```diff
@@ -600,6 +600,13 @@ case ${step} in
         [[ "${DO_ICE}" == "YES" ]] && nthreads_cice6=1
       fi
 
+      if [[ ${ntiles} -gt 6 ]] ; then
```
Comment (aerorahul) [inline parm/config/gfs/config.resources]:
```suggestion
      if (( ntiles > 6 )); then
```

File: parm/config/gfs/config.ufs [hunk 3]
```diff
@@ -68,6 +68,21 @@ if [[ "${skip_mom6}" == "false" ]] || [[ "${skip_cice6}" == "false" ]] || [[ "${
   skip_mediator=false
 fi
 
+if [[ "${DO_NEST:-NO}" == YES ]] ; then
```
Comment (aerorahul) [inline parm/config/gfs/config.ufs]:
```suggestion
if [[ "${DO_NEST:-NO}" == "YES" ]] ; then
    # Describe nest location, interaction with parent, etc.
```

File: parm/config/gfs/config.ufs [hunk 4]
```diff
@@ -105,6 +120,24 @@ case "${fv3_res}" in
...
+            export nest_joffset=9
+            export npx_nest=361
+            export npy_nest=241
+            export NEST_DLON=0.25
+            export NEST_DLAT=0.25
+            export WRITE_GROUP=2
+            export WRTTASK_PER_GROUP_PER_THREAD_PER_TILE=2
+            export WRITE_GROUP_GFS=2
+            export WRTTASK_PER_GROUP_PER_THREAD_PER_TILE_GFS=2
+        fi
```
Comment (aerorahul) [inline parm/config/gfs/config.ufs]: If you could please cast this as:
```bash
if [[ "${DO_NEST:-NO}" == "YES" ]]; then
   settings with nest
else
    settings without nest
fi
```
that would help ensuring they are grouped appropriately

File: ush/forecast_postdet.sh [hunk 5]
```diff
@@ -175,6 +175,10 @@ EOF
       if [[ "${WRITE_DOPOST}" == ".true." ]]; then
         ${NLN} "${COM_ATMOS_MASTER}/${RUN}.t${cyc}z.master.grb2f${FH3}" "GFSPRS.GrbF${FH2}"
         ${NLN} "${COM_ATMOS_MASTER}/${RUN}.t${cyc}z.sfluxgrbf${FH3}.grib2" "GFSFLX.GrbF${FH2}"
+        if [[ "${DO_NEST:-NO}" == YES ]] ; then
+            ${NLN} "${COM_ATMOS_MASTER}/${RUN}.t${cyc}z.nest.grb2f${FH3}" "GFSPRS.GrbF${FH2}.nest02"
```
Comment (aerorahul) [inline ush/forecast_postdet.sh]: The indent is 2 spaces in this script.  Please try to stay consistent.
4 spaces is fine, but when the strings are long, every little bit helps with readability.

File: ush/forecast_postdet.sh [hunk 6]
```diff
@@ -196,7 +200,12 @@ FV3_nml() {
   source "${USHgfs}/parsing_namelists_FV3.sh"
   source "${USHgfs}/parsing_model_configure_FV3.sh"
 
-  FV3_namelists
+  if [[ "${DO_NEST:-NO}" == YES ]] ; then
+      FV3_namelists global_with_nest
+      FV3_namelists nest
+  else
+      FV3_namelists #default to global without nest
+  fi
```
Comment (aerorahul) [inline ush/forecast_postdet.sh]:
```suggestion
  # Call the appropriate namelist functions
  if [[ "${DO_NEST:-NO}" == YES ]] ; then
      FV3_namelists global_with_nest
      FV3_namelists nest
  else
      FV3_namelists
  fi
```

File: parm/config/gfs/config.base [hunk 7]
```diff
@@ -184,6 +184,13 @@ export DO_AERO="NO"
 export WAVE_CDUMP="" # When to include wave suite: gdas, gfs, or both
 export DOBNDPNT_WAVE="NO"
 export FRAC_GRID=".true."
+export DO_NEST="NO" # Whether to run a global-nested domain
+if [[ "${DO_NEST:-NO}" == "YES" ]] ; then
+  export ntiles=7
+  export NEST_OUTPUT_GRID="regional_latlon"
+else
+  export ntiles=6
```
Comment (aerorahul) [inline parm/config/gfs/config.base:192]: It probably is a good idea to put `export ntiles=6` in `parm/config/gefs/config.base` as well as `export DO_NEST="NO"`

File: parm/config/gfs/config.fcst [hunk 8]
```diff
@@ -253,7 +253,11 @@ export DO_SPPT=${DO_SPPT:-"NO"}
 export DO_SKEB=${DO_SKEB:-"NO"}
 export DO_SHUM=${DO_SHUM:-"NO"}
 export DO_LAND_PERT=${DO_LAND_PERT:-"NO"}
-export DO_CA=${DO_CA:-"YES"}
+if [[ "${DO_NEST:-NO}" == YES ]] ; then
+  export DO_CA="NO" #CA does not work with nesting. 
+else
+  export DO_CA=${DO_CA:-"YES"}
+fi
```
Comment (aerorahul) [inline parm/config/gfs/config.fcst]:
```suggestion
export DO_CA=${DO_CA:-"YES"}
if [[ "${DO_NEST:-NO}" == "YES" ]] ; then
  export DO_CA="NO"  # CA does not work with nesting.
fi
```

File: parm/config/gfs/config.stage_ic [hunk 9]
```diff
@@ -38,4 +38,8 @@ case "${CASE}" in
     ;;
 esac
 
+if [[ "${DO_NEST:-NO}" == "YES" ]] ; then
+  export CPL_ATMIC=GLOBAL-NEST_${CASE}
```
Comment (aerorahul) [inline parm/config/gfs/config.stage_ic]:
```suggestion
  export CPL_ATMIC="GLOBAL-NEST_${CASE}"
```

File: ush/parsing_namelists_FV3.sh [hunk 10]
```diff
@@ -6,6 +6,20 @@
 # shellcheck disable=SC2034
 FV3_namelists(){
 
+# First argument tells us which namelist we're writing:
```
Comment (aerorahul) [inline ush/parsing_namelists_FV3.sh]: We are in the process of replacing this with using a template from the ufs-weather-model, and this change makes it very difficult for us to do that at this time.
I will discuss with @junwang-noaa to find out if there is a way to do it before this part of the PR.

**Review by aerorahul — CHANGES_REQUESTED** [review]
Most of the changes look good.
There are some minor changes requested.
2 major comments:
1. `model_configure` template should be used from ufs-weather-model rather than `cat` 
2. `FV3_namelists` is about to be replaced with a much smaller, leaner method that uses `input.nml.IN` from ufs-weather-model.  The changes in this PR make it difficult to address.  We should discuss offline to come up with a mitigation strategy so that it does not prevent either of us to do the things we want.

File: parm/config/gfs/config.base (hunk 7 above)
Comment (guoqing-noaa) [reply parm/config/gfs/config.base:192]: Sure. Thanks!

File: parm/config/gfs/config.ufs (hunk 4 above)
Comment (guoqing-noaa) [reply parm/config/gfs/config.ufs]: Sure. Thanks!

File: ush/forecast_postdet.sh (hunk 5 above)
Comment (guoqing-noaa) [reply ush/forecast_postdet.sh]: Sure. Thanks!

File: ush/parsing_namelists_FV3.sh (hunk 10 above)
Comment (guoqing-noaa) [reply ush/parsing_namelists_FV3.sh]: Sounds good. Thanks!

File: ush/parsing_model_configure_FV3.sh (hunk 1 above)
Comment (guoqing-noaa) [reply ush/parsing_model_configure_FV3.sh]: Thanks! I will make necessary changes here and create a PR to the ufs-weather-model as well.

**Comment by guoqing-noaa** [comment]
> Most of the changes look good. There are some minor changes requested. 2 major comments:
> 
> 1. `model_configure` template should be used from ufs-weather-model rather than `cat`
> 2. `FV3_namelists` is about to be replaced with a much smaller, leaner method that uses `input.nml.IN` from ufs-weather-model.  The changes in this PR make it difficult to address.  We should discuss offline to come up with a mitigation strategy so that it does not prevent either of us to do the things we want.

@aerorahul Thanks a lot for your great help and great suggestions. I think I addressed most comments except `FV3_namelists`. 
Are you good that I go ahead to set up a tag up meeting with you and relevant parties on this?

**Comment by guoqing-noaa** [comment]
FYI, PR https://github.com/ufs-community/ufs-weather-model/pull/2252 was created to add tests/parm/model_configure_nest.IN

**Review by aerorahul — APPROVED** [review]
Looks good.
Thanks for accepting the suggestions and working them in.
Just a note:  
A model update will be needed to find the `model_configure_nest.IN` at some point.  Until then, the `DO_NEST=YES` feature will need to use a ufs-weather-model hash provided by someone else.

**Comment by guoqing-noaa** [comment]
Thanks, @aerorahul !

Yes, at the moment, we will manually put in place the `model_configure_nest.IN` file when we run the global nest workflow.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests.

**Comment by guoqing-noaa** [comment]
Thanks, @WalterKolczynski-NOAA and @aerorahul !

**Pull request merged**